src/System.Reflection.Emit/tests/MethodBuilder/MethodBuilderMakeGenericMethod.cs

```diff
@@ -0,0 +1,167 @@
+            Type returnType = typeof(int);
+            Type[] desiredType = new Type[] {
+                typeof(string),
+                typeof(int)
+            };
+
+            MethodBuilder builder = TestTypeBuilder.DefineMethod(methodName,
+                TestMethodAttributes);
+            Type[] typeParameters =
+                builder.DefineGenericParameters(new string[] { "T", "U" }).Select(a => a.AsType()).ToArray();
+            builder.SetParameters(typeParameters);
+            builder.SetReturnType(returnType);
+            MethodInfo methodInfo = builder.MakeGenericMethod(desiredType);
+            Assert.True(VerifyMethodInfo(methodInfo, builder, returnType));
+        }
+
+        [Fact]
+        public void NegTest1()
+        {
+            string methodName = null;
+            methodName = TestLibrary.Generator.GetString(false, false, true, MinStringLength, MaxStringLength);
+            Type returnType = typeof(void);
+
+            MethodBuilder builder = TestTypeBuilder.DefineMethod(methodName,
+                TestMethodAttributes);
+            Type[] typeParameters =
+                builder.DefineGenericParameters(new string[] { "T" }).Select(a => a.AsType()).ToArray();
+            builder.SetParameters(typeParameters);
+            builder.SetReturnType(returnType);
+
+            MethodInfo methodInfo = builder.MakeGenericMethod(null);
+        }
+
+        [Fact]
+        public void NegTest2()
+        {
+            string methodName = null;
+            methodName = TestLibrary.Generator.GetString(false, false, true, MinStringLength, MaxStringLength);
+            Type returnType = typeof(void);
+
+            MethodBuilder builder = TestTypeBuilder.DefineMethod(methodName,
+                TestMethodAttributes);
+            Type[] typeParameters =
+                builder.DefineGenericParameters(new string[] { "T" }).Select(a => a.AsType()).ToArray();
+            builder.SetParameters(typeParameters);
+            builder.SetReturnType(returnType);
+
+            MethodInfo methodInfo = builder.MakeGenericMethod(new Type[] { });
+        }
+
+        [Fact]
+        public void NegTest3()
+        {
+            string methodName = null;
+            methodName = TestLibrary.Generator.GetString(false, false, true, MinStringLength, MaxStringLength);
+            Type returnType = typeof(void);
+
+            MethodBuilder builder = TestTypeBuilder.DefineMethod(methodName,
+                TestMethodAttributes);
+            Type[] typeParameters =
+                builder.DefineGenericParameters(new string[] { "T" }).Select(a => a.AsType()).ToArray();
+            builder.SetParameters(typeParameters);
+            builder.SetReturnType(returnType);
+
+            MethodInfo methodInfo = builder.MakeGenericMethod();
+        }
+
+        private bool VerifyMethodInfo(MethodInfo methodInfo, MethodBuilder builder, Type returnType)
```

Comment:
Same comment as on the other verification method (asserting instead of returning bools).  I'll refrain from commenting on these further if there are any more.